CONTRIBUTING.md

```diff
@@ -47,55 +57,83 @@ If you want to build tar and zip files for distribution (they will also appear i
+3. For a basic demo, visit `http://localhost:8000/test/demo/basic.html`
 
-    grunt test
+You can also get a report of the tests' code coverage:
 
-To test a specific module only (`navigator` here):
+    grunt coverage
 
-    grunt test --module="navigator"
+The report shows up at `coverage/html/index.html` viewable in a browser.
 
-> [!TIP]
-> The argument can be found
-> for example in the module definition: `QUnit.module('<name here>', ...`
 
-If you wish to work interactively with the tests or test your changes:
+### Installing from forked GitHub repo/branch
 
-    grunt connect watch
+This project is now compatible with direct installation of forked Github repos/branches via npm/yarn (possible because of the new [prepare](https://docs.npmjs.com/misc/scripts) command).  This enables quick testing of a bugfix or feature addition via a forked repo.  In order to do this:
 
-and open `http://localhost:8000/test/test.html` in your browser.
+1. Install the Grunt command line runner (if you haven't already); on the command line, run `npm install -g grunt-cli` (or `yarn global add grunt-cli`)
+1. Remove any currently installed openseadragon package via `npm uninstall openseadragon` or `yarn remove openseadragon`
+1. Add the specific forked repo/branch by running `npm install git://github.com/username/openseadragon.git#branch-name` or `yarn add git://github.com/username/openseadragon.git#branch-name`. Make sure to replace username and branch-name with proper targets.
 
-Another good page, if you want to interactively test out your changes, is `http://localhost:8000/test/demo/basic.html`.
+During installation, the package should be correctly built via grunt and can then be used via `import Openseadragon from 'openseadragon'` or `var Openseadragon = require('openseadragon')` statements as if the official package were installed.
 
+### Pull Request Workflow
 
-> [!NOTE] 
-> corresponding npm commands for the above are:
->  - npm run test
->  - npm run test -- --module="navigator"
->  - npm run dev
+When you're ready to contribute your changes:
 
-You can also get a report of the tests' code coverage:
+1. **Fork the repository** if you haven't already
+1. **Create a branch** for your feature or bugfix (`git checkout -b my-new-feature`)
+1. **Make your changes** following the code style guidelines
+1. **Run the tests** to ensure nothing breaks (`grunt test`)
+1. **Commit your changes** (`git commit -am 'Add some feature'`)
+1. **Push to your branch** (`git push origin my-new-feature`)
+1. **Create a Pull Request** from your fork to the main repository
 
-    grunt coverage
+Please provide a clear description in your pull request that explains:
+- What the change does
+- Why it's needed
+- How it's been tested
+- Any related issues (use "Fixes #123" or "Relates to #123" syntax)
 
-The report shows up at `coverage/html/index.html` viewable in a browser.
 
-### Installing from forked Github repo/branch
+### Troubleshooting
```

Comment:
I'm not convinced we need the Troubleshooting section at all. Everything in here feels like generic advice, not something relevant specifically to this project. Are there specific items in here that you think are particularly important?